mantidproject/mantid
# Notify Batch Presenter on Preview Apply

**Description of work.**
This PR connects the Apply button on the Preview tab to the Experiment Settings tab. When applying a change in the ROI's for a specific angle, it should be updated in the Experiment settings.

**Questions before finished**
- Do we want a g_log help message, or a popup message? And if popup, how is this done in Refl?
     _We will use g_log inside of the PreviewPresenter for now, and then later we will pass the IMessageHandler to the PreviewPresenter_
- Is it possible to have multiple rows with matching angles/titles? And what do we want to do in this case?
      _It is possible to have multiple rows. For now we will just throw and catch if there is multiple of the same. The PreviewPresenter will catch this so we can show a popup at some stage_
- The `PreviewRow` doesn't appear to have a title member variable, so how do we match this to a `LookupRow`, or do we want to just search by angle like its currently doing?
      _Ignore the title field for now. If two rows with the same angle are found, then throw_
- How do we want to test the ExperimentPresenter? Do we want to mock the model out, or create a fake Experiment object with some data? 
     _A separate issue has been opened to handle this, as both options proved to be complex and require further consideration_

**To test:**

Must be tested in a debug build and have the ISIS archive enabled.

 - Ensure this workspace is loaded or is in your load path: [INTER45455_inst.zip](https://github.com/mantidproject/mantid/files/7504266/INTER45455_inst.zip)
 - Open the ISIS Reflectometry GUI.
- Select the Preview tab.
- Type in INTER45455_inst, set the angle to 1, and click Load. The instrument view plot should display the data.
- Select a region of interest on the slice viewer.
- Click the Apply button.
- Select the Experiment Settings tab and check that the ROI values from the Preview tab have been entered into the wildcard row there.
- Test that this works for the different types of ROI - signal, transmission and background should all be entered into the relevant columns in the Experiment Settings table.
- Enter a row with an angle into the table on the Experiment Settings tab. Check that entering a matching angle on the Preview tab and clicking Apply causes the selected ROIs to be entered into the matching row in the Experiment Settings table.

Part of #34240

*This does not require release notes* because **this part of Refl GUI is still under development**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
